src/storage.ts

```diff
@@ -0,0 +1,266 @@
+// Storage utility types
+enum StorageAccessLevel {
```

Comment:
any reason of not using a union type instead?

The only reason Auth uses enums in some types is to align with the other platforms, but idiomatically speaking a union type offer a best DX IMO.

Reply:
Hum I suppose I prefer the DX of an enum, but I could see arguments for both. I do think we should be consistent where possible. Let me think on this.

Reply:
I'm in camp union types over enums, however our team / app has traditionally preferred enums. Enums are okay exposed across api boundaries as long as they have their values defined like is done here. If memory serves they are worse on memory, speed due to their added complexity.

There are many articles/posts ranting about their issues, but having read several, it mostly comes down to having sharp edges and being easy to misunderstand to bad effect.

DX-wise, when I open quote on a string union field, I get a list of the valid strings in my autocomplete. With an enum, I think I can use string equivalents, but I always feel like I need to import the enum and fully reference the available values.